src/Idris/Parser.hs

```diff
@@ -711,9 +712,10 @@ postulate syn = do (doc, ext)
                                    ext <- ppostDecl
                                    return (doc, ext)
                    ist <- get
-                   let initOpts = if default_total ist
-                                     then [TotalFn]
-                                     else []
+                   let initOpts = case default_total ist of
+                                    DefaultCheckingTotal    -> [TotalFn]
+                                    DefaultCheckingCovering -> [CoveringFn]
+                                    _                       -> []
                    opts <- fnOpts initOpts
```

Comment:
I think the `fnOpts` removes the `initOpts` if an explicit totality is given.
However this now works incorrectly since `CoveringFn` would not be removed if you look at the code.

This hack has seemingly been there for a long time and I think it would be nice if you managed to fix it. I would propose removing `initOpts` from `fnOpts` and instead checking after parsing the options whether there exists an annotation and if not adding the default one (if there exist multiple, it should fail which for some reason or other it doesn't do now).